lib/langchain/tool/calculator.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+module Langchain::Tool
+  #
+  # A calculator tool that falls back to the Google calculator widget
+  #
+  # Gem requirements:
+  #     gem "eqn", "~> 1.6.5"
+  #     gem "google_search_results", "~> 2.0.0"
+  #
+  # Usage:
+  #     calculator = Langchain::Tool::Calculator.new
+  #
+  class Calculator
+    extend Langchain::ToolDefinition
+    include Langchain::DependencyHelper
+
+    define_action :execute, description: "Evaluates a pure math expression or if equation contains non-math characters (e.g.: \"12F in Celsius\") then it uses the google search calculator to evaluate the expression" do
```

Comment:
This produces the follow function name:
```ruby
langchain_tool_calculator__execute
```
I'm wondering whether it should be simpler, just the class name without the namespace:
```ruby
calculator__execute
```

Reply:
Yes, we could do something like:

```ruby
define_action :execute, for: "calculator", description: "..." do
  # ...
end
```

The "__execute" part would be added automatically based on the method name.

However, I have some doubts about this approach. It feels a bit odd to define the tool name for each function. You'd have to repeat the same name multiple times, and if you make a typo, you lose consistency.

Since we're already inside the tool class, I would expect Langchain to handle this automatically.

We could add another method like:

```ruby
define_tool :calculator
```

This way, you only write the name once. But it's an extra step to remember.

I prefer Langchain to automatically define the name, but if we have to choose between these two options, I would lean towards the second one, which is similar to the old approach.

Reply:
Maybe what we do is this:
```ruby
class Langchain::Tool::Calculator
  define_action :execute

  def execute()
```
becomes `name: "langchain_tool_calculator__execute"`

But you can also have an option to override the name completely with:
```ruby
class Langchain::Tool::Calculator
  define_action :execute, name: "calc__execute"

  def execute()
```
becomes `name: "calc__execute"`

What do you think?

Reply:
After a discussion with @andreibondarev, it has been decided to keep the current functionality unchanged for now.